src/unix/socket.c

```diff
@@ -496,13 +496,10 @@ closure_function(1, 2, sysreturn, unixsock_close,
     unixsock s = bound(s);
     unixsock_lock(s);
     if (s->conn_q) {
-        /* Deallocate any sockets in the connection queue. */
+        /* Close any sockets in the connection queue. */
         unixsock child;
-        while ((child = dequeue(s->conn_q)) != INVALID_ADDRESS) {
-            unixsock_lock(child);
-            unixsock_dealloc(child);
-        }
-
+        while ((child = dequeue(s->conn_q)) != INVALID_ADDRESS)
+            fdesc_close_internal(t->p, &s->sock.f, s->sock.fd);
```

Comment:
It's the child socket that should be closed, so the second and third arguments should be &child->sock.f and child->sock.fd.
An alternative approach to solve the problem would be to not allocate a file descriptor when a socket is enqueued in conn_q, but only if/when it's dequeued during an accept().

Reply:
Ok, I changed the code to defer allocating an fd until the enqueued socket is accepted. I removed fdesc_close_internal() as there is no need for it now.